### PR TITLE
[Android Components Incubator] TabCounter 

### PR DESCRIPTION
**Pull request by cnevinc**
Ready for review,
After landed, will test it on Release channel and request to merge into Android-components

@pocmo Please help advise. Thanks!

### CODE REVIEW

**Comment by cnevinc** [comment]
APK size 爆惹... QQ

**Comment by mTwTm** [comment]
I took a look at the components size difference between 10 and 16, no big difference. I think the oversized apk is because you did not minifyEnabled false in ui-tabcounter/build.gradle?

File: settings.gradle [hunk 1]
```diff
@@ -1,4 +1,4 @@
-include ':feature-tabs'
+include ':feature-tabs', ':ui-tabcounter'
```
Comment (walkingice) [inline settings.gradle]: separate into 2 lines?

File: app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java [hunk 2]
```diff
@@ -82,6 +81,9 @@
 import java.lang.ref.WeakReference;
 import java.util.WeakHashMap;
 
+import incubator.org.mozilla.components.ui.tabcounter.TabCounter;
+
+
```
Comment (walkingice) [inline app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java]: redundant new line

File: settings.gradle (hunk 1 above)
Comment (walkingice) [inline settings.gradle]: separate to this?
```
include ':feature-tabs'
include ':ui-tabcounter'
```

File: app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java (hunk 2 above)
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java]: fixed

**Comment by pocmo** [comment]
> Don't use utils from a-c to make Rocket APK smaller

@cnevinc That's interesting. I would have expected proguard to remove everything except the single helper method you use. Do you remember how big the difference is?

**Comment by cnevinc** [comment]
Firstly I have the same thought as you. But I look at the APK difference I found kotlinx/coroutine and more kotlin classes I didn't use. 
I'm still investigating 
image

**Pull request closed without merging**